core/controllers/galleries.py

```diff
@@ -95,6 +95,37 @@ def get(self):
         })
         self.render_template('galleries/gallery.html')
 
+class GallerySearchPage(base.BaseHandler):
+    """The exploration gallery search page."""
+
+    PAGE_NAME_FOR_CSRF = 'gallery'
+
+    def get(self):
+        """Handles GET requests."""
+        gallery_handler = GalleryHandler(self.request, self.response)
+        gallery_handler.get_data()
+        explorations_list = gallery_handler.values['explorations_list']
+        query_response = self.request.get('q').split()
+        query_string_list = [x for x in query_response
```

Comment:
I don't understand what's going on here. Can you explain? (Why are we modifying the query string?)